fg21sim/extragalactic/pointsources/pointsources.py

```diff
@@ -0,0 +1,361 @@
+            step = ps_radius / 10  # Should be fixed
+            # x and y are the differencial rad to the core point at the theta and
+            # phi directions.
+            x = np.arange(-ps_radius, ps_radius + step, step) * au.rad
+            y = np.arange(- ps_radius,  ps_radius + step, step) * au.rad
+            for p in range(len(x)):
+                for q in range(len(y)):
+                    if np.sqrt(x[p].value**2 + y[q].value**2) <= ps_radius:
+                        x_ang = (x[p].to(au.deg) + theta).value / 180 * np.pi
+                        y_ang = (y[q].to(au.deg) + phi).value / 180 * np.pi
+                        if x_ang > np.pi:
+                            x_ang -= np.pi
+                        elif x_ang < 0:
+                            x_ang += np.pi
+                        if y_ang > 2 * np.pi:
+                            y_ang -= 2 * np.pi
+                        elif y_ang < 0:
+                            y_ang += 2 * np.pi
+                        pix_tmp = hp.ang2pix(
+                            self.nside, x_ang, y_ang)
+                        hpmap[pix_tmp] += ps_flux_list[i]
+
+        return hpmap
+
+    def draw_lobe(self, ps_data, ps_type, freq):
+        """
+        Designed to draw the elliptical lobes of FRI and FRII
+
+        Prameters
+        ---------
+        nside: int and dyadic
+        ps_data: pandas.core.frame.DataFrame
+            Data of the point sources
+        ps_type: int
+            Class type of the point soruces
+        freq: float
+            frequency
+        """
+        # Init
+        npix = hp.nside2npix(self.nside)
+        hpmap = np.zeros((npix,))
+        num_ps = ps_data.shape[0]
+        # Gen flux list
+        ps_flux_list = self.calc_flux(ps_type, freq, ps_data)
+        ps_lobe = ps_flux_list[:, 1]
+        # Iteratively draw ps
+        for i in range(num_ps):
+            # Parameters
+            theta = ps_data['Theta (deg)'][i] * au.deg
+            phi = ps_data['Phi (deg)'][i] * au.deg
+            lobe_maj = ps_data['lobe_maj (rad)'][i]
+            lobe_min = ps_data['lobe_min (rad)'][i]
+            lobe_ang = ps_data['lobe_ang (deg)'][i] / 180 * np.pi
+
+            # Lobe1
+            lobe1_theta = lobe_maj * np.cos(lobe_ang) * au.rad
+            lobe1_theta = theta + lobe1_theta.to(au.deg)
+            lobe1_phi = lobe_maj * np.sin(lobe_ang) * au.rad
+            lobe1_phi = phi + lobe1_phi.to(au.deg)
+            # Focuses
+            lobe_c = np.sqrt(lobe_maj**2 - lobe_min**2)  # focus distance
+            F1_core_x = lobe_c
+            F1_core_y = 0
+            F2_core_x = -lobe_c
+            F2_core_y = 0
+            # draw
+            step = lobe_maj / 10
+            x = np.arange(-lobe_maj, lobe_maj + step, step)
+            y = np.arange(-lobe_min, lobe_min + step, step)
+            # Ellipse
+            for p in range(len(x)):
+                for q in range(len(y)):
+                    DistFocus1 = np.sqrt(
```

Comment:
Use ``numpy`` array manipulations as much as possible, and avoid explicit Python loops as much as possible.